### PR TITLE
statistics: add metrics for unneeded analyze table

### LINKED ISSUE #54823: add metrics for unneeded analyze
## Enhancement

We currently monitor the health of various tables, but among those with low health, there are tables that do not require triggering ANALYZE. Therefore, merely looking at tables with low health does not reveal whether the ```ANALYZE``` speed is sufficient or if there will be a backlog of tasks.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #54823

Problem Summary:

### What changed and how does it work?

 
The primary changes in ```pkg/statistics/handle/cache/statscache.go``` file.
the rest are code migrations to avoid cyclic imports.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

I test in the 1m tables cluster. we can see this metrics in the grafana.

image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/statistics/table.go [hunk 1]
```diff
@@ -709,6 +713,19 @@ func (t *Table) IsAnalyzed() bool {
 	return t.LastAnalyzeVersion > 0
 }
 
+// IsEligibleForAnalysis checks whether the table is eligible for analysis.
+func (t *Table) IsEligibleForAnalysis() bool {
+	// 1. If the statistics are either not loaded or are classified as pseudo, there is no need for analyze.
+	//	  Pseudo statistics can be created by the optimizer, so we need to double check it.
```
Comment (hi-rustin) [inline pkg/statistics/table.go:719]:
```suggestion
	// 1. If the statistics are either not loaded or are classified as pseudo, there is no need for analyze.
	//    Pseudo statistics can be created by the optimizer, so we need to double check it.
```

**Review by hi-rustin — COMMENTED** [review]
Do we need to update the grafana josn file?

**Comment by hawkingrei** [comment]
> Do we need to update the grafana json file?

No, because this type is generate dynamically by then grafana.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**